lidarrmetadata/provider.py

```diff
@@ -462,8 +463,125 @@ def search_artist(self, name):
         return [{'Id': result.get_mbid(),
                  'Overview': result.get_bio_summary()}
                 for result in results]
+    
+class MusicbrainzSearchProvider(Provider,
+                                ArtistNameSearchMixin):
+    
+    """
+    Provider that uses the Musicbrainz external lucene search (i.e. not postgres)
+    """
+    def __init__(self,
+                 search_server='http://musicbrainz.org'):
+        """
+        Class initialization
+
+        Defaults to the offical musicbrainz webservice but in principle we could
+        host our own mirror using https://github.com/metabrainz/musicbrainz-docker
+
+        :param search_server: URL for the search server.  Note that using HTTPS adds around 100ms to search time.
+        """
+        super(MusicbrainzSearchProvider, self).__init__()
 
+        self._search_server = search_server
+        self._limiter = _get_rate_limiter(key='musicbrainz_search')
+        
+        self._stats = stats.TelegrafStatsClient(CONFIG.STATS_HOST,
+                                                CONFIG.STATS_PORT) if CONFIG.ENABLE_STATS else None
+
+            
+    def _count_request(self, result_type):
+        if self._stats:
+            self._stats.metric('external', {result_type: 1}, tags={'provider': 'musicbrainz_search'})
+        
+    def get_with_limit(self, url):
+        
+        try:
+            with self._limiter.limited():
+                self._count_request('request')
+                return requests.get(url, timeout=CONFIG.EXTERNAL_TIMEOUT)
+        except HTTPError as error:
+            logger.error('HTTPError: {e}'.format(e=error))
+            return {}
+        except requests.exceptions.Timeout as error:
+            logger.error('Timeout: {e}'.format(e=error))
+            self._count_request('timeout')
+            return {}
+        except limit.RateLimitedError:
+            logger.error('Musicbrainz search request rate limited')
+            self._count_request('ratelimit')
+            return {}
+        
+    def search_artist_name(self, name, limit=None, albums=[]):
+        
+        if albums:
+            return self.search_artist_name_with_albums(name, albums, limit)
 
+        query = self.escape_lucene_query(name)
+        
+        url = u'{server}/ws/2/artist/?query={query}&fmt=json&offset=0&dismax=true'.format(server=self._search_server, query=query)
+        
+        if limit:
+            url += u'&limit={}'.format(limit)
+        
+        response = self.get_with_limit(url)
+        print("Search for {query} completed in {time}ms".format(query=name, time=response.elapsed.microseconds / 1000))
+        
+        return self.parse_artist_search(response.json())
```

Comment:
Probably want to check `response.status_code == 200` and optionally record a stat for response code.